src/middlewares/auth.js

```diff
@@ -4,81 +4,84 @@ import models from '../db/models';
 const { User } = models;
 
 const auth = {
-    verifyToken(token) {
-        let decoded = {};
-        try {
-            decoded.payload = jwt.verify(token, process.env.JWT_SECRET);
-        } catch (error) {
-            decoded = { error: error.message };
-        }
-        return decoded;
-    },
+  verifyToken(token) {
+    let decoded = {};
+    try {
+      decoded.payload = jwt.verify(token, process.env.JWT_SECRET);
+    } catch (error) {
+      decoded = { error: error.message };
+    }
+    return decoded;
+  },
 
-    async verifyUserToken(req, res, next) {
-        try {
-            let token;
+  async verifyUserToken(req, res, next) {
+    try {
+      let token;
 
-            if (req.headers.authorization) {
-                token = req.headers.authorization;
-            } else if (req.headers['x-access-token']) {
-                token = req.headers['x-access-token'];
-            } else if (req.headers.token) {
-                token = req.headers.token;
-            }
+      if (req.headers.authorization) {
+        token = req.headers.authorization;
+      } else if (req.headers['x-access-token']) {
+        token = req.headers['x-access-token'];
+      } else if (req.headers.token) {
+        token = req.headers.token;
+      }
 
-            const decoded = auth.verifyToken(token);
+      const decoded = auth.verifyToken(token);
 
-            if (!token) {
-                return res.status(401).json({
-                    status: 'error',
-                    error: 'No token provided!',
-                });
-            }
+      if (!token) {
+        return res.status(401).json({
+          status: 'error',
+          error: 'No token provided!',
+        });
+      }
 
-            if (decoded.error) {
-                return res.status(401).json({
-                    status: 'error',
-                    error: 'Failed to authenticate token.',
-                });
-            }
+      if (decoded.error) {
+        return res.status(401).json({
+          status: 'error',
+          error: 'Failed to authenticate token.',
+        });
+      }
 
-            req.currentUser = await User.findOne({where: {id: decoded.payload.id}});
-            return next();
-        } catch (error) {
-            return res.status(500).json({
-                status: 'error',
-                error: ('Internal Server Error'),
-            });
-        }
-    },
+      const user = await User.findOne({ where: { id: decoded.payload.id } });
+      req.currentUser = user;
+      const { payload } = decoded;
+      req.tokenData = payload;
+      return next();
+    } catch (error) {
+      return res.status(500).json({
```

Comment:
FOR FUTURE PURPOSES: Let's try to pass 500 errors to the application's default handler like so:4

```
try {
...
}catch(error) { next(error) }

```

This will make our code more concise. 
Again, YOU DON"T HAVE TO EFFECT THIS CHANGE NOW.